### PR TITLE
Upgrade version of Microsoft.Azure.WebJobs.Extensions.Http to remove vulnerability

### PR DESCRIPTION
**Pull request by jackbatzner**
There is a vulnerability in Microsoft.AspNetcore.Http in 2.1.0 that we need to upgrade

References:
- https://github.com/advisories/GHSA-hxrm-9w7p-39cc
- https://github.com/dotnet/aspnetcore/pull/24264

### CODE REVIEW

**Comment by jackbatzner** [comment]
Hi @justinyoo  - Please take a peek at this PR, there's a high vulnerability that should be resolved for consumers of this library.

File: src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.csproj [hunk 1]
```diff
@@ -26,7 +26,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.0.12" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.2.0" />
     <PackageReference Include="Microsoft.OpenApi" Version="1.2.3" />
     <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
```
Comment (justinyoo) [inline src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.csproj:31]: We also need to change this `Newtonsoft.Json` version to `13.0.1`, and all other projects referring to the version of `11.0.2` need to change its version to `13.0.1`

File: samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc.csproj [hunk 2]
```diff
@@ -17,6 +17,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.2.0" />
```
Comment (justinyoo) [inline samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc.csproj:20]: We don't need this

File: samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc.csproj [hunk 3]
```diff
@@ -17,6 +17,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.1.0" />
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />
```
Comment (justinyoo) [inline samples/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc/Microsoft.Azure.WebJobs.Extensions.OpenApi.FunctionApp.InProc.csproj:19]: Instead we need to change this `Microsoft.NET.Sdk.Functions` to `4.1.2`

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />
+    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.Http" Version="3.2.0" />
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj:13]: We don't need this

File: test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj [hunk 5]
```diff
@@ -10,6 +10,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="4.0.1" />
```
Comment (justinyoo) [inline test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp/Microsoft.Azure.WebJobs.Extensions.OpenApi.TestApp.csproj:12]: Instead we need to change this `Microsoft.NET.Sdk.Functions` to `4.1.2`

**Review by justinyoo — CHANGES_REQUESTED** [review]
Thanks @jackbatzner for this PR. I've left some comments for you to take a look.

File: src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Extensions/TypeExtensions.cs [hunk 6]
```diff
@@ -459,7 +459,18 @@ public static Type GetUnderlyingType(this Type type)
...
+                {
+                    if (type.IsNested)
+                    { // Does not work for implemented interfaces
+                        underlyingType = type.BaseType;
+                    }
+                    else
+                    {
+                        underlyingType = type.GetGenericArguments()[0];
+                    }                    
+                }
```
[the remark this answers is not in the document]
Comment (jackbatzner) [reply src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Extensions/TypeExtensions.cs]: Yup! This was a mistake. I thought I had pulled those changes to a different branch

**Review by justinyoo — CHANGES_REQUESTED** [review]
LGTM! Thanks for the PR

**Review by justinyoo — APPROVED** [review]

**Pull request merged**